api/src/main/java/org/apache/iceberg/actions/AnalyzeTable.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+import java.util.List;
+import java.util.Set;
+
+/** An action that collects statistics of an Iceberg table and writes to Puffin files. */
+public interface AnalyzeTable extends Action<AnalyzeTable, AnalyzeTable.Result> {
+  /**
+   * The set of columns to be analyzed
+   *
+   * @param columnNames a set of column names to be analyzed
+   * @return this for method chaining
+   */
+  AnalyzeTable columns(String... columnNames);
```

Comment:
if we gonna support multiple different blob types (`types(Set<String)`), should the `columns` be specified for them separately? For example i could want to calculate NDV for col A and B, and correlation blob type for {B,C} together.

i think we should replace `types(Set<String>)` and `columns(String...)` with just one method:

```
type(String type, Set<String> columns)
```

(not sure about the name, it should be cumulative)

Reply:
> Though we take type as input in the API, we can restrict the usage by not exposing it in the procedure?

yes, that's definitely an option

Reply:
> > correlation blob type for {B,C} together
> 
> This is the stats on the combined value of B and C. Is my understanding right? Is this a common usecase since i didnt find databases supporting this by default

Optimizers (like Trino CBO) assume column independence and thus can highly skew estimates when faced with correlated data. I hope we solve this problem some day. See also https://www.singlestore.com/blog/correlation-statistics-in-singlestore/

Reply:
Actually thinking about this, i feel its better in subsequent pr, especially after we consider more types in the spec.  Its currently hard to imagine the use case.  So , let's remove the code to configure and check types in the API, as it does not make sense with one type.  

I am leaning towards using a metrics-style config for columns, for configuring the types per column.  (ie, write.metadata.stats.column.a.types)

Reply:
> So , let's remove the code to configure and check types in the API, as it does not make sense with one type.

sgtm